src/source_sampling.cpp

```diff
@@ -125,6 +127,65 @@ pyne::Sampler::Sampler(std::string filename,
       bias_tag_name = tag_names["bias_tag_name"];
     }
   }
+  if (tag_names.find("e_bounds_tag_name") == tag_names.end()) {
+    // e_bounds not provided by h5m
+    if (e_bounds.size() == 0) {
+      throw std::invalid_argument("e_bounds_tag_name not found");
+    }
+  } else { // e_bounds provided in h5m file
+    if (e_bounds.size() > 0) {
+      // there is also an user defined e_bounds
+      std::cout<<"warning: e_bounds in 'source.h5m' will be used."<<std::endl;
+    }
+  }
+  setup();
+}
+
+pyne::Sampler::Sampler(std::string filename,
+                 std::map<std::string, std::string> tag_names,
```

Comment:
Indentation updates needed here.

Reply:
Done in https://github.com/pyne/pyne/pull/1341/commits/0315a7f4cde0e9080c8d4f77b27e4a5f3665a7e3.